### DISCUSSION TITLE
Is it possible for wfb-ng to configure a mavlink port (NOT telemetry)?

### DISCUSSION THREAD

**Discussion by lida2003** [svpcom/wfb-ng#366] [Questions] [answered]
Is it possible for wfb-ng to configure a tunnel for pulling [IMU data](https://github.com/ArduPilot/ardupilot/pull/27007) at frequency above 100Hz?

The bandwith (from FC to CC) might be over (14 x 2 + mavlink head(10B)) x 8bit/B x 100 / 1 second > 30 400 bps

**Comment by svpcom** [comment]
If you want to pack every IMU reading into separate packet then it will eat a lot of bandwidth.
If do some aggregation then it should work.
Why you want to use tunnel for it? Just use ordinary mavlink channel (it already has aggregation)

**Reply by lida2003** [reply]
I'm thinking about using my Jetson Orin(on the ground) for Non-GPS position, which is normally on the copter connecting FC through UART.

I need these IMU data and video at the same time, and it says IMU data frequency should be large than 100Hz. So I hope to evaluate those figures first. And it seems not optimistic.

--- EDIT:

The entire packet from FC is consturcted here in ardupilot:

https://github.com/ArduPilot/ardupilot/blob/0dbe9ed27fb28bfeee4063b5cf2634b851e6a690/libraries/GCS_MAVLink/GCS_Common.cpp#L2141-L2214

OpenIPC will use https://github.com/OpenIPC/mavfwd to handle mavlink packet, which might inject mavlink packet into thoese video frame (I think, but i don't know detals yet.) Is there any test on bandwidth or estimated numbers?

**Reply by svpcom** [reply, marked as answer]
OpenIPC use only subset of wfb-ng stack.
WFB-ng (full version) do mavlink aggregation out of box: https://github.com/svpcom/wfb-ng/wiki/Mavlink-mode

**Reply by lida2003** [reply]
I read below links:
1. https://github.com/svpcom/wfb-ng/wiki/Setup-HOWTO 
2. https://github.com/svpcom/wfb-ng/wiki/Mavlink-mode

From `Mavlink-mode`, it seems if I wanna use >100Hz rate, then `mavlink_agg_timeout = 0.01`, which has been mentioned above cause eating a lot of bandwidth of the video.

>By default WFB-NG encapsulates one source UDP packet into one WiFi packet. But mavlink packets are very small (usually less than 100 bytes) and send them in separate packets produces too much overhead. So for mavlink packets there is default setting [common] mavlink_agg_timeout = 0.1 which will aggragate mavlink packets while they less than radio_mtu but no longer than 100ms.

And I also noticed that there is a `IPoWB`, I'm NOT sure what it is. It seems not the MAVLink port passthrough to 14550, and it not recommended for telemetry streams, just for ssh. Which migh be some developer for remote login the CC.

> For IPoWB (IPv4 over Wifibroadcast tunnel) you need only tun/tap kernel driver (tun.ko). Tunnel will be configured and work out of box. Drone side will have 10.5.0.2/24 address and GS - 10.5.0.1/24. Please note that tunnel use less efficient coding rate (to minimize latency) than video and mavlink streams and use it only for low-bandwidth traffic (like ssh), not as general link for video and telemetry streams.

**Reply by svpcom** [reply]
Do you really need 100Hz? If yes, then `mavlink_agg_timeout = 0.01` is a only solution and it will issue small wifi packets.
IPoWB (ip tunnel) will work the same.

**Reply by lida2003** [reply]
Thanks. I'll try to see what I will get. vins-mono suggest to use > 100Hz, so let's find out if it works :)
